metadatastore/commands.py

```diff
@@ -65,7 +65,7 @@ def reconfigure(self, config):
     def _connection(self):
         if self.__conn is None:
             self.__conn = MongoClient(self.config['host'],
-                                      self.config.get('port', None))
+                                      int(self.config.get('port', None)))
```

Comment:
`int(None)` will raise.  `None` is a valid option for the port in MongoClient.__init__.  Therefore I think this code needs to be:

```python
if self.__conn is None:
    port = self.config.get('port')
    if port:
        port = int(port)
    self.__conn = MongoClient(self.config['host'], port)
```

or

```python
 int(self.config.get('port', 27017)))
```